easypy/randutils.py

```diff
@@ -42,6 +47,49 @@ def get_chunks(self, offset, end, block_size_range, shuffle=False):
             offset += size
 
 
+def load_dictionary(filename):
+    full_path = os.path.join(os.path.dirname(__file__), 'dictionary', filename)
+    with open(full_path) as dict_file:
+        dictionary = [line.strip() for line in dict_file]
+
+    grouped_by_len = grouped(dictionary, key=len)
+    return collections.defaultdict(list, grouped_by_len)
+
+
+NOUNS = load_dictionary('nouns.txt')
+VERBS = load_dictionary('verbs.txt')
+ADVERBS = load_dictionary('adverbs.txt')
+ADJECTIVES = load_dictionary('adjectives.txt')
+PARTS_OF_SPEECH = (NOUNS, VERBS, ADVERBS, ADJECTIVES)
+
+
+def random_meaningful_name(max_length=50, sep='-'):
+    parts = []
+    remaining = max_length
+
+    if max_length < 3:
+        return random_string(max_length, charset=string.ascii_lowercase)
+
+    for part_group in PARTS_OF_SPEECH:
+        max_part_len = remaining - len(sep) if parts else max_length
```

Comment:
let's do a test run; you the first three words with length (x, y, z respectively) and the separator '-' (length is 1) with max_length as 50 that means the remaining = 50 - x - 1 - y - 1 - z - 1, now you reach here and what to know how long your string can be, well parts is not empty so it's True, hence max_part_len will be remaining - 1 but there is no separator at the end of the last part, so there is no reason to decrease its size from max_part_len of the last part